tools/pipelines/templates/include-upload-npm-logs.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# Template to upload logs from npm (not pnpm!) invocations during a job run.
+# These can be useful for debugging npm-related issues
+
+parameters:
+  # Path to npm logs directory.
+  # Defaults to the path that npm ends up using in Azure Pipelines if there are no overrides to its config.
+  # Files that include this template generally shouldn't need to change this.
+  - name: npmLogsDirectory
+    type: string
+    default: '/home/cloudtest/.npm/_logs'
+
+steps:
+- task: Bash@3
+  displayName: Upload npm logs
+  condition: succeededOrFailed()
+  continueOnError: true
+  inputs:
+    targetType: 'inline'
+    workingDirectory: '${{ parameters.npmLogsDirectory }}'
+    script: |
+      set -eu -o pipefail
```

Comment:
You know. :) can we externalize this script?

Reply:
I was starting to but I think I lean towards skipping in this case... the last line wouldn't make sense in an extracted script since it's very specifically tied to ADO pipelines. The rest of the script is basically running `tar`, so not sure it's worth extracting.